resolwe/flow/executors/run.py

```diff
@@ -0,0 +1,322 @@
+        return functools.reduce(get_item, k.split('.'), d)
+
+    else:
+        # Set value
+        try:
+            k, k_last = k.rsplit('.', 1)
+            set_item(dict_dot(d, k, default=dict), k_last, val)
+        except ValueError:
+            set_item(d, k, val)
+        return val
+
+
+# to ensure uniform access, wrap the json settings into this, because
+# the code below will only access them in a django-like fashion (through attributes, not dict items)
+settings = DictObjBridge(json.load(open('settings.json', 'rt')))  # pylint: disable=invalid-name
+
+now = datetime.datetime.now  # pylint: disable=invalid-name
+
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+CWD = os.getcwd()
+
+
+def iterjson(text):
+    """Decode JSON stream."""
+    decoder = json.JSONDecoder()
+    while len(text) > 0:
+        obj, ndx = decoder.raw_decode(text)
+
+        if not isinstance(obj, dict):
+            raise ValueError()
+
+        text = text[ndx:].lstrip('\r\n')
+        yield obj
+
+
+class BaseFlowExecutor(object):
+    """Represents a workflow executor."""
+
+    exported_files_mapper = defaultdict(dict)
+
+    def __init__(self, *args, **kwargs):
+        """Initialize attributes."""
+        self.data_id = None
+        self.process = None
+        # Flag to mark current object as failed.
+        self.requirements = {}
+        self.resources = {}
+        self.redis = redis.StrictRedis(**settings.FLOW_EXECUTOR.get('REDIS_CONNECTION', {}))
+        self.manager_cookie = 1
+        self.queue_response_channel = '{}.{}'.format(settings['REDIS_CHANNEL_PAIR'][1], settings['Data_get']['id'])
+
+    def get_tools(self):
+        """Get tools paths."""
+        tools_paths = settings.FLOW_EXECUTOR_TOOLS_PATHS
+        return tools_paths
+
+    def start(self):
+        """Start process execution."""
+        pass
+
+    def run_script(self, script):
+        """Run process script."""
+        raise NotImplementedError('`run_script` function must be defined')
+
+    def end(self):
+        """End process execution."""
+        pass
+
+    def get_stdout(self):
+        """Get process' standard output."""
+        return self.stdout  # pylint: disable=no-member
+
+    def _send_manager_command(self, cmd, expect_reply=True, **kwargs):
+        """Send a properly formatted command to the manager."""
+        packet = {
+            'id': settings['Data_get']['id'],
+            'action': cmd,
+            'cookie': self.manager_cookie,
+        }
+        packet.update(kwargs)
+
+        # TODO what happens here if the push fails? we don't have any realistic recourse,
+        # so just let it explode and stop processing
+        queue_channel = settings['REDIS_CHANNEL_PAIR'][0]
+        try:
+            self.redis.rpush(queue_channel, json.dumps(packet))
+        except:  # pylint: disable=bare-except
+            logger.error("Error sending command to manager:\n\n%s", traceback.format_exc())
```

Comment:
Why don't you use `logger.exception` here?

Reply:
Also where are these things logged anyway?

Reply:
At the moment they are not logged anywhere. We should discuss what is the best way to collect them.
Do we want to log them locally on the executor? How to support services like Sentry in this case?
Do we want to somehow send them back to master and put them into central logging system?

Anyway I think that it is out of the scope of this pull request and will be added later. For now it is enough to print them on stdout or stderr, but this probably already happens as the error is re-raised.